packages/core/src/domain/conversion/bundle-modifications/__tests__/post-process-bundle.test.ts

```diff
@@ -0,0 +1,225 @@
+  const documentExtension = buildDocIdFhirExtension(fileName);
+  return { patientId, condition, fileName, documentExtension };
+}
+
+describe("Checking postProcessBundle and its constituent functions", () => {
+  describe("replaceIDs", () => {
+    it("does not update the IDs if the extension is missing", () => {
+      const { patientId, condition } = initTest();
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId);
+      const updCondition = updatedBundle.entry?.[0]?.resource;
+      expect(updCondition).toBeDefined();
+      expect(updCondition?.id).toBe(condition.id);
+      expect(updCondition).toMatchObject(condition);
+    });
+
+    it("updates the ID and adds source metadata when extension is present", () => {
+      const { patientId, condition, fileName, documentExtension } = initTest();
+      condition.extension = [documentExtension];
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId);
+      const updCondition = updatedBundle.entry?.[0]?.resource;
+      expect(updCondition).toBeDefined();
+      expect(updCondition?.id).toBeDefined();
+      expect(updCondition?.id).not.toBe(condition.id);
+      expect(updCondition?.meta?.source).toBe(fileName);
+      expect(updCondition).toMatchObject({
+        ...condition,
+        id: updCondition?.id,
+        meta: { source: fileName },
+      });
+    });
+
+    it("throws error when bundle entry is empty", () => {
+      const { patientId } = initTest();
+      const bundle = makeBundle({ entries: [], type: "collection" });
+      expect(() => bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId)).toThrow(
+        "Missing bundle entries"
+      );
+    });
+
+    it("throws error when bundle has no entries", () => {
+      const { patientId } = initTest();
+      const bundle = makeBundle({ type: "collection" });
+      expect(() => bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId)).toThrow(
+        "Missing bundle entries"
+      );
+    });
+  });
+
+  describe("addExtensionToConversion", () => {
+    it("adds a document extension to a resource that did not have it", () => {
+      const { condition, documentExtension } = initTest();
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.addExtensionToConversion(bundle, documentExtension);
+      const updCondition = updatedBundle.entry?.[0]?.resource as Condition | undefined;
+      expect(updCondition).toBeDefined();
+      if (!updCondition) throw new Error("Bundle entry got removed!");
+
+      expect(updCondition.extension).toHaveLength(1);
+      expect(updCondition.extension?.[0]).toEqual(documentExtension);
+      expect(condition.extension).toBeUndefined();
```

Comment:
Do expect `addExtensionToConversion` to modify `bundle`, or is this a check for the initialization - in which case, can we move it to before the call to `addExtensionToConversion` so its clearer this is not part of the actual test, just a pre-condition to it, please?

Reply:
makes sense. Moved

Reply:
Actually, I put that there to make sure that the source object didn't get altered